aws-janitor/resources/internet_gateways.go

```diff
@@ -77,6 +77,37 @@ func (InternetGateways) MarkAndSweep(opts Options, set *Set) error {
 				break
 			}
 
+			var publicIPsToRelease []*string
+
+			pageFunc := func(page *ec2.DescribeNetworkInterfacesOutput, _ bool) bool {
+				for _, eni := range page.NetworkInterfaces {
+					publicIPsToRelease = append(publicIPsToRelease, eni.Association.PublicIp)
```

Comment:
I believe association can be nil, causing cleanup to panic in certain cases. This also made difficult by code in main() that appears to hide panic logging as it calls explicit os.Exit.

Also, deleting network interfaces should disassociate automatically, so likely we don't need this change - I'm suspecting that ENIs are held by untagged ELBs.